projects/igniteui-angular/src/lib/data-operations/filtering-util.ts

```diff
@@ -0,0 +1,59 @@
+import { IFilteringExpression } from './filtering-expression.interface';
+import { FilteringExpressionsTree, IFilteringExpressionsTree } from './filtering-expressions-tree';
+
+export class FilteringUtil {
+    /**
+     * Returns the filtering expression for a column with the provided tree and fieldName.
+     * ```typescript
+     * let filteringExpression = FilteringUtil.find(gridExpressionTree, 'Column Field');
+     * ```
+     */
+    public static find(tree: IFilteringExpressionsTree, fieldName: string): IFilteringExpressionsTree | IFilteringExpression {
+        const index = this.findIndex(tree, fieldName);
+
+        if (index > -1) {
+            return tree.filteringOperands[index];
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns the index of the filtering expression for a column with the provided tree and fieldName.
+     * ```typescript
+     * let filteringExpressionIndex = FilteringUtil.findIndex(gridExpressionTree, 'Column Field');
+     * ```
+     */
+    public static findIndex(tree: IFilteringExpressionsTree, fieldName: string): number {
+        let expr;
+        for (let i = 0; i < tree.filteringOperands.length; i++) {
+            expr = tree.filteringOperands[i];
+            if (expr instanceof FilteringExpressionsTree) {
```

Comment:
Since we touched on the subject - this is still technically a circular reference as `FilteringExpressionsTree` depends on `FilteringUtil` (runtime) and vice versa, so the imports between the two modules remain. I suppose since usage is in methods, the order is not important and that might be why the tooling doesn't complain.
Regardless, what's more is that the actual class `FilteringExpressionsTree` is not supposed to be manadatory so these really shouldn't be checking for instance like that. Perhaps a simple check for `filteringOperands` and the rest of these can remain as pure types (can even be explicitly imporeted as such)?

Reply:
And noticed `expr` isn't really typed, but if you want to do so or this could also just be for-of instead - the check can be with a type guard function to narrow the type for the inner call and skip the assert on the else :)

Reply:
I'm a bit wary of modifying the original code, but I've changed the way the type checks are done, and removed the `FilteringExpressionsTree` requirement.